tiledb/sm/config/test/unit_config.cc

```diff
@@ -32,76 +32,108 @@
 
 #define CATCH_CONFIG_MAIN
 #include <test/support/tdb_catch.h>
+#include "../../misc/parse_argument.h"
 #include "../config.h"
 
 using tiledb::sm::Config;
+using tiledb::sm::utils::parse::convert;
 
 /*
  * These are the types for which `Config::get` is explicitly specialized.
  */
 using config_types = std::
     tuple<bool, int, uint32_t, int64_t, uint64_t, float, double, std::string>;
-using config_types_integral = std::tuple<int, uint32_t, int64_t, uint64_t>;
-using config_types_floating = std::tuple<float, double>;
+/*
+ * Config types that are convertable by utils::parse::convert
+ */
+using config_types_convertable =
+    std::tuple<bool, int, uint32_t, int64_t, uint64_t, float, double>;
 
-TEST_CASE("Config::get - not found", "[config]") {
-  Config c{};  // empty
-  std::string key{"this_key_is_not_present"};
-  auto x{c.get(key)};
-  CHECK(!x.has_value());
+TEMPLATE_LIST_TEST_CASE(
+    "Config::get<T> - found", "[config]", config_types_convertable) {
+  Config c{};
+  std::string key{"the_key"};
+  CHECK(c.set(key, "1").ok());
+  REQUIRE_NOTHROW(c.get<TestType>(key));
+  auto found_value{c.get<TestType>(key)};
+  CHECK(found_value.has_value());
+  TestType expected_value;
+  CHECK(convert("1", &expected_value).ok());
+  CHECK(found_value.value() == expected_value);
```

Comment:
This test isn't actually testing what it ought to be. Calling `convert` here is just replicating the algorithm it's testing, which means it's not actually a test.. The expected values should not be converted but instead present in the test vector.

I'll admit Catch doesn't make this easy. You have to do it with specializations and `constexpr`.